src/main/java/de/tum/in/www1/artemis/domain/hestia/ExerciseHint.java

```diff
@@ -1,19 +1,31 @@
-package de.tum.in.www1.artemis.domain;
+package de.tum.in.www1.artemis.domain.hestia;
 
 import javax.persistence.*;
 
 import org.hibernate.annotations.Cache;
 import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.annotations.DiscriminatorOptions;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.Exercise;
 
 /**
- * A ExerciseHint.
+ * An ExerciseHint.
  */
 @Entity
 @Table(name = "exercise_hint")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
```

Comment:
In a previous implementation, `ExerciseHint` was abstract and `TextHint` (the current ExerciseHint) and `CodeHint` extended ExerciseHint in our model and code.
We discussed with Stephan whose opinion it was to change this to our current implementation.

Reply:
To add to Oles comment:
This was decided because CodeHints now have all the attributes that TextHints have with the addition of being linked to SolutionEntries. Before they did not have the content attribute, but this was changed after discussing it with Stephan.